### PR TITLE
Implement Group Policy settings for PSResource Repository

### LINKED ISSUE #1696: Group Policy for a PSRepository Allow List/Registered List
### Summary of the new feature / enhancement

The group policy will register repositories as the only repositories allowed to be installed from if the policy is enabled. 
If the policy is enabled no other repositories will be allowed. 

The policy will take in a list of repositories, at minimum a name;uri pair or a full hashtable (or json may be preferred) of repository information. 

The policy will initially be Windows only but we will explore enabling on non-Windows via Intune policy and file system. 

The policy will be checked at each call to a repository, we will assume same privileges for all operations. 

There will also be a function/api that will expose if the policy is enabled and what repository Uris are available so that other tools which call PSResourceGet (such as WinGet) can integrate to this functionality. 

If an exact repo is specified that is not allowed it will be a terminating error. If no repo is specified all allowed repos will be seared and it will be a non-terminating error. 

### Proposed technical implementation details (optional)

image

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Implement Group Policy settings for PSResource Repository. 
Policy allows administrator to define an allow list for Uris that are allowed to be used for finding and installing resources.

The format for the policy is as follows:

`Name=PSGallery;Uri=https://www.powershellgallery.com/api/v2/`

## PR Context

Fixes #1696 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [x] [Documentation needed]()
        - [X] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs-PSGet/issues/163
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: doBuild.ps1 [hunk 1]
```diff
@@ -34,9 +34,19 @@ function DoBuild
     Copy-Item -Path "./LICENSE" -Dest "$BuildOutPath"
 
     # Copy notice
-    Write-Verbose -Verbose -Message "Copying ThirdPartyNotices.txt to '$BuildOutPath'"
+    Write-Verbose -Verbose -Message "Copying InstallPSResourceGetPolicyDefinitions.ps1 to '$BuildOutPath'"
```
Comment (alerickson) [inline doBuild.ps1]: Should this verbose logging be added to line 39 instead (ie leave the verbose messaging for coping the notice and add a new verbose statement for copying policy definitions)?

Comment (alerickson) [reply doBuild.ps1]: Oh I see it already exists on line 41, so this statement should be reverted to what it previously was (verbose message for ThirdPartyNotices.txt)

File: src/code/ServerFactory.cs [hunk 2]
```diff
@@ -1,11 +1,12 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.PowerShell.PSResourceGet.UtilClasses;
+using System;
 using System.Collections;
+using System.Linq;
```
Comment (alerickson) [inline src/code/ServerFactory.cs]: Are these 2 additions still needed?

**Review by alerickson — APPROVED** [review]

File: doBuild.ps1 (hunk 1 above)
Comment (adityapatwardhan) [reply doBuild.ps1]: Updated it to be correct. `Notice.txt`

File: src/code/ServerFactory.cs (hunk 2 above)
Comment (adityapatwardhan) [reply src/code/ServerFactory.cs]: Removed all unnecessary

**Pull request merged**